File: app/src/main/java/com/intelegain/agora/utils/Sharedprefrences.java
package com.intelegain.agora.utils;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * This is SharedPreferences SingleTon Class used to store small amount of data
 * in private mode
 */
public class Sharedprefrences {
    private static Sharedprefrences instance;
    private SharedPreferences pref;
    private SharedPreferences.Editor editor;
    private final String name = "AgoraIntelgain";

    /**
     * @param context Parameterized Constructor is called
     */
    public static Sharedprefrences getInstance(Context context) {
        return instance == null ? instance = new Sharedprefrences(context) : instance;
    }

    /**
     * Default Constructor To make this class work as SingleTon
     */
    private Sharedprefrences() {
    }

    /**
     * Constructor is called
     */
    private Sharedprefrences(Context context) {
        pref = context.getSharedPreferences(name, 0);
        editor = pref.edit();
    }

    /**
     * Here all values is to put into shared prefrences
     *
     * @param key   Unique Key for a value
     * @param value Value to be stored
     */
    public void putboolean(String key, boolean value) {
        editor.putBoolean(key, value);
        editor.commit();
    }

    public void putString(String key, String value) {
        editor.putString(key, value);
        editor.commit();
    }

    public void putInt(String key, int value) {
        editor.putInt(key, value);
        editor.commit();
    }

    public void putLong(String key, long value) {
        editor.putLong(key, value);
        editor.commit();
    }

    public void putFloat(String key, float value) {
        editor.putFloat(key, value);
        editor.commit();
    }

    public void remove(String key) {
        editor.remove(key);


    }

    public void clear() {
        editor.clear();
        //editor.commit();


    }

    public void commit() {
        editor.commit();
    }

    /**
     * Here all values is to get String value from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found defalutValue is assigned
     * @return string
     */
    public String getString(String key, String defaultvalue) {
        return pref.getString(key, defaultvalue);
    }

    /**
     * Here all values is to get int value from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found defalutValue is assigned
     * @return int
     */
    public int getInt(String key, int defValue) {
        return pref.getInt(key, defValue);
    }

    /**
     * Here all values is to get boolean value from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found defalutValue is assigned
     * @return boolean
     */
    public boolean getBoolean(String key, boolean defValue) {
        return pref.getBoolean(key, defValue);
    }

    /**
     * Here all values is to get long valued from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found defalutValue is assigned
     * @return long
     */
    public long getLong(String key, long defValue) {
        return pref.getLong(key, defValue);
    }

    /**
     * Here all values is to get float value from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found
     *                     defalutValue is assigned
     * @return float
     */
    public float getFloat(String key, float defValue) {
        return pref.getFloat(key, defValue);
    }
}

File: app/src/main/java/com/intelegain/agora/utils/Sharedprefrences.kt
package com.intelegain.agora.utils

import android.content.Context
import android.content.SharedPreferences

/**
 * This is SharedPreferences SingleTon Class used to store small amount of data
 * in private mode
 */
class Sharedprefrences {
    private lateinit var pref: SharedPreferences
    private var editor: SharedPreferences.Editor? = null
    private val name = "AgoraIntelgain"

    /**
     * Default Constructor To make this class work as SingleTon
     */
    private constructor() {}

    /**
     * Constructor is called
     */
    private constructor(context: Context?) {
        if (context != null) {
            pref = context.getSharedPreferences(name, 0)
        }
        editor = pref.edit()
    }

    /**
     * Here all values is to put into shared prefrences
     *
     * @param key   Unique Key for a value
     * @param value Value to be stored
     */
    fun putboolean(key: String?, value: Boolean) {
        editor!!.putBoolean(key, value)
        editor!!.commit()
    }

    fun putString(key: String?, value: String?) {
        editor!!.putString(key, value)
        editor!!.commit()
    }

    fun putInt(key: String?, value: Int) {
        editor!!.putInt(key, value)
        editor!!.commit()
    }

    fun putLong(key: String?, value: Long) {
        editor!!.putLong(key, value)
        editor!!.commit()
    }

    fun putFloat(key: String?, value: Float) {
        editor!!.putFloat(key, value)
        editor!!.commit()
    }

    fun remove(key: String?) {
        editor!!.remove(key)
    }

    fun clear() {
        editor!!.clear()
        //editor.commit();
    }

    fun commit() {
        editor!!.commit()
    }

    /**
     * Here all values is to get String value from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found defalutValue is assigned
     * @return string
     */
    fun getString(key: String?, defaultvalue: String?): String? {
        return pref!!.getString(key, defaultvalue)
    }

    /**
     * Here all values is to get int value from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found defalutValue is assigned
     * @return int
     */
    fun getInt(key: String?, defValue: Int): Int {
        return pref!!.getInt(key, defValue)
    }

    /**
     * Here all values is to get boolean value from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found defalutValue is assigned
     * @return boolean
     */
    fun getBoolean(key: String?, defValue: Boolean): Boolean {
        return pref!!.getBoolean(key, defValue)
    }

    /**
     * Here all values is to get long valued from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found defalutValue is assigned
     * @return long
     */
    fun getLong(key: String?, defValue: Long): Long {
        return pref!!.getLong(key, defValue)
    }

    /**
     * Here all values is to get float value from shared preferences
     *
     * @param key          retrieve by unique key
     * @param defaultvalue give here defaultValue if not found
     * defalutValue is assigned
     * @return float
     */
    fun getFloat(key: String?, defValue: Float): Float {
        return pref!!.getFloat(key, defValue)
    }

    companion object {
        private var instance: Sharedprefrences? = null
        /**
         * @param context Parameterized Constructor is called
         */
        @JvmStatic
        fun getInstance(context: Context?): Sharedprefrences {
            return if (instance == null) Sharedprefrences(context).also { instance = it } else instance!!
        }
    }
}